mlir/Makefile

```diff
@@ -74,6 +74,7 @@ dialects:
 		-DQUANTUM_ENABLE_BINDINGS_PYTHON=ON \
 		-DPython3_EXECUTABLE="$(PYTHON)" \
 		-DMLIR_DIR=$(LLVM_BUILD_DIR)/lib/cmake/mlir \
+		-DMHLO_DIR=$(MHLO_BUILD_DIR)/lib/cmake/mlir-hlo \
```

Comment:
One more additional dependency to build dir :cry:

Reply:
This is a net win. The end result of this is that we won't need the `mlir-hlo-opt` binary.

Reply:
I'm personally interested in reducing such dependencies by relying on prebuilt artifacts and upstreaming extensions to these projects if possible. Anyway, I'm not against this one now.

Reply:
Yes, I agree. I think there's potential to upstream some changes. We can chat about them some time soon :)

Reply:
@maliasadi do you have any more comments on this?

Reply:
All good!